tests/bluetooth/host_config_variants/testcase.yaml

```diff
@@ -0,0 +1,16 @@
+tests:
+  # Test that the Host builds with BT_SMP_OOB_LEGACY_PAIR_ONLY enabled
+  bluetooth.host_config_variants.config_bt_smp_oob_legacy_pair_only:
```

Comment:
My suggestion is to also add a test with
```
CONFIG_BT_CENTRAL=y
CONFIG_BT_SMP_OOB_LEGACY_PAIR_ONLY=y
```
And for both peripheral and central tests add:
`CONFIG_COMPILER_OPT="-Werror"`
to ensure that the warnings are no longer there

Reply:
Test are already build with warning as error. Please, if what I did is not enough, create an issue.

As pointed by @jhedberg I am not sure https://github.com/zephyrproject-rtos/zephyr/pull/74400 was the best way to go to fix the issue. Also it would have been nice to have a regression test added to avoid the current situation...